### PR TITLE
Add OpenBLAS 0.3 Cmake Config

### PR DESCRIPTION
**Pull request by xsacha**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -267,3 +268,25 @@ if(PKG_CONFIG_FOUND)
...
+export(TARGETS ${OpenBLAS_LIBNAME}
+FILE ${CMAKE_BINARY_DIR}/${TARGETS_EXPORT_NAME}.cmake)
+
+install(TARGETS ${OpenBLAS_LIBNAME}
+        EXPORT ${TARGETS_EXPORT_NAME}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+configure_file("cmake/Config.cmake.in"
```
Comment (ruslo) [inline CMakeLists.txt]: Use `configure_package_config_file` here since we need `check_required_components`.

File: cmake/Config.cmake.in [hunk 2]
```diff
@@ -0,0 +1,3 @@
+include(CMakeFindDependencyMacro)
+find_dependency(Threads)
+include("${CMAKE_CURRENT_LIST_DIR}/@TARGETS_EXPORT_NAME@.cmake")
```
Comment (ruslo) [inline cmake/Config.cmake.in:5]: We should use `check_required_components` here even if there are no components:
* https://github.com/cgold-examples/fruits/blob/a4b78e26ae3a6c7244174ae869f3447f36871eb4/cmake/template/Config.cmake.in#L4
* https://gitlab.kitware.com/cmake/cmake/merge_requests/2138

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (xsacha) [reply CMakeLists.txt]: Ah, I had that originally then I looked at how Opencl did it and thought this way was simpler for less dependencies.
I'll make that change tomorrow I guess.

File: cmake/Config.cmake.in (hunk 2 above)
Comment (xsacha) [reply cmake/Config.cmake.in:5]: Should be good now

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/OpenBLAS/releases/tag/v0.3.0-p1

**Pull request merged**